### ISSUE TITLE
Improve warnings for syntax issues in transform yaml

### ISSUE THREAD

**Issue by annelo-msft** [Azure/autorest#4270]
I was receiving the following error for the following transform yaml:

```
``` yaml
directive:
    from:swagger-document
    where: $.paths["/v2/{name}/manifests/{reference}"].put
    transform: >
      $.parameters.push({
            "name": "Content-Type",
            "in": "header",
            "type": "string"
          });
```

![image](https://user-images.githubusercontent.com/21269347/131191911-5cd82414-c4c9-4bc5-a858-2891319c039e.png)

It wasn't clear what was wrong, but it turns out I was missing a space after the `:` in `from:swagger-document`.  This works just fine:

```
``` yaml
directive:
    from: swagger-document
    where: $.paths["/v2/{name}/manifests/{reference}"].put
    transform: >
      $.parameters.push({
            "name": "Content-Type",
            "in": "header",
            "type": "string"
          });
```

It would be great if we could improve warnings around syntax errors to help people debug swagger transforms more easily.  Thanks!

**Comment by timotheeguerin** [comment]
I actually been doing some work to improve the logger and fix those types of error not showing the information. You should be able to test with the @next version of core(and might need to try the @next version of the cli as well)
Otherwise those should be published soon(maybe next week)

**Comment by timotheeguerin** [comment]
Closing as it should be fixed. Please reopen if this is still present in the next version or when it is published (`@autorest/core 3.6.x` and `autorest 3.4.x`)
